fcrepo-clients/src/main/java/edu/unc/lib/dl/util/VocabularyHelperManager.java

```diff
@@ -86,6 +86,9 @@
 	private Boolean initialized = false;
 
 	public synchronized void init() {
+		// Wait for the repository to be up before loading vocabularies
+		managementClient.waitForRepositoryAvailable();
```

Comment:
It is probably needed in any code that will not be running in tomcat, such
as the services workers (in fact, it is needed in there), so I think it
makes sense to have the check for repository availability in
fcrepo-clients.  But having something that blocked startup of the deposit
service until fedora came online would actually be wise.  I can't recall if
the attempt to get the Collections object pid fails if fedora is offline,
it may be going directly to mulgara and therefore not care.

On Tue, Aug 18, 2015 at 9:56 AM, daines <notifications@github.com> wrote:

> In
> fcrepo-clients/src/main/java/edu/unc/lib/dl/util/VocabularyHelperManager.java
> <https://github.com/UNC-Libraries/Carolina-Digital-Repository/pull/354#discussion_r37300262>
> :
>
> > @@ -86,6 +86,9 @@
> >  	private Boolean initialized = false;
> >
> >  	public synchronized void init() {
> > +		// Wait for the repository to be up before loading vocabularies
> > +		managementClient.waitForRepositoryAvailable();
>
> Since other parts of the deposit service depend on the availability of the
> repository, could this be moved into DepositSupervisor?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UNC-Libraries/Carolina-Digital-Repository/pull/354/files#r37300262>
> .
>